### PR TITLE
add more tests for basic functions

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/test/test-alloc-util.c [hunk 1]
```diff
@@ -37,20 +39,55 @@ static void test_alloca(void) {
...
 
         assert_se(dup);
         assert_se(dup[0] == 1);
         assert_se(dup[1] == 2);
         assert_se(dup[2] == 3);
-        free(dup);
+}
+
+static void test_memdup_suffix0_multiply(void) {
+        uint8_t org[] = {1, 2, 3};
```
Comment (keszybz) [inline src/test/test-alloc-util.c]: For the test to be more useful, this shouldn't be uint8_t, but something with a bigger size.

File: src/test/test-alloc-util.c [hunk 2]
```diff
@@ -37,20 +39,55 @@ static void test_alloca(void) {
...
+}
+
+static void test_greedy_realloc0(void) {
+        _cleanup_free_ int *buf;
+        int *p;
+        size_t i, allocated = 3;
+
+        buf = new(int, allocated);
+        assert_se(buf);
+        assert_se(buf[0] = 1);
```
Comment (keszybz) [inline src/test/test-alloc-util.c]: I think those assert_se's here are unnecessary.

**Comment by yuwata** [comment]
@keszybz Thanks. test-alloc-util is updated. Also some tests for capability set are added.

**Comment by keszybz** [comment]
`sudo: unable to resolve host testgreen`

**Pull request merged**